tests/ui/new_without_default.rs

```diff
@@ -201,4 +201,21 @@ pub mod issue7220 {
     }
 }
 
+// see issue #8152
+// This should not create any lints
+pub struct DocHidden;
+impl DocHidden {
+    #[doc(hidden)]
+    pub fn new() -> Self {
+        DocHidden
+    }
+}
+
+pub struct DocShown;
+impl DocShown {
+    pub fn new() -> Self {
+        DocShown
+    }
+}
```

Comment:
This is just the base test case and is already tested in this file. No need to test this again.

Reply:
Totally overlooked that. Thanks!

Reply:
Pushed changes to the PR. Thanks for the review!